server/utils/mojPagination/pagination.ts

```diff
@@ -0,0 +1,87 @@
+import { ApiPageRequest, ApiPageResponse } from '../../data/ApiData'
+import PageResponse from './pageResponse'
+
+type PageLink = {
+  text: string
+  href: string
+}
+
+type NumberedPageLink = {
+  text: string
+  href: string
+  selected?: boolean // default = false
+}
+
+type ResultsMetaData = {
+  from: number
+  to: number
+  count: number
+}
+
+type PageViewModel = {
+  results: ResultsMetaData
+  previous: PageLink | null
+  next: PageLink | null
+  items: NumberedPageLink[] | null
+}
+
+export function pageRequestFrom(pageSize: number, pageNumber: number): ApiPageRequest {
+  return {
+    size: pageSize,
+    number: pageNumber - 1,
+  }
+}
+
+export default function mojPaginationFromPageResponse<T>(
+  reportPageResponse: ApiPageResponse<T>,
+  url: URL
+): PageViewModel {
+  const pageResponse = new PageResponse(
+    reportPageResponse.size,
```

Comment:
Is there any sense in having the PageResponse take an ApiPageResponse here instead of the individual fields. Only reason I say that is that it then makes it harder to mess up in other bits of the code with the indexing maybe?